google/cloud/spanner/internal/time_format.cc

```diff
@@ -34,6 +37,62 @@ namespace spanner {
 inline namespace SPANNER_CLIENT_NS {
 namespace internal {
 
+namespace {
+
+constexpr auto kDigits = "0123456789";
+
+// Formats [0 .. 99] as %02d.
+inline char* Format02d(char* ep, int v) {
+  *--ep = kDigits[v % 10];
+  *--ep = kDigits[(v / 10) % 10];
+  return ep;
+}
+
+char const* ParseInt(char const* dp, int min, int max, int* vp) {
+  int value = 0;
+  bool neg = false;
+  if (*dp == '-') {
+    neg = true;
+    ++dp;
+  }
+  char const* const bp = dp;
+  constexpr int kMin = std::numeric_limits<int>::min();
+  while (char const* cp = std::strchr(kDigits, *dp)) {
```

Comment:
> I am guessing this explicit look is here because std::from_chars() is unavailable to us, and std::strtol() is too slow? Consider a comment explaining why.

Added such a comment.